src/EditorFeatures/Core/IntelliSense/AsyncCompletion/ItemManager.CompletionListUpdater.cs

```diff
@@ -529,50 +577,72 @@ private ImmutableArray<CompletionFilterWithState> GetUpdatedFilters(IReadOnlyLis
+
+                    // Most recently used item is our top preference.
+                    if (currentResultMruIndex != bestResultMruIndex)
+                    {
+                        if (currentResultMruIndex > bestResultMruIndex)
+                        {
+                            bestResult = currentResult;
+                            bestResultMruIndex = currentResultMruIndex;
+                        }
+
+                        continue;
+                    }
+
+                    // 2nd preference is IntelliCode item
+                    var currentIsPreferred = currentResult.CompletionItem.IsPreferredItem();
+                    var bestIsPreferred = bestResult.CompletionItem.IsPreferredItem();
 
-                    if (currentItemMruIndex > bestItemMruIndex ||
-                        (currentItemMruIndex == bestItemMruIndex && !bestItem.IsPreferredItem() && currentItem.IsPreferredItem()))
+                    if (currentIsPreferred != bestIsPreferred)
                     {
-                        bestItem = currentItem;
-                        bestItemMruIndex = currentItemMruIndex;
+                        if (currentIsPreferred && !bestIsPreferred)
+                        {
+                            bestResult = currentResult;
+                        }
+
+                        continue;
                     }
-                }
 
-                // If our best item appeared in the MRU list, use it
-                if (_recentItemsManager.GetRecentItemIndex(bestItem) >= 0)
-                {
-                    return bestItem;
-                }
+                    // 3rd preference is higher MatchPriority
+                    var currentMatchPriority = currentResult.CompletionItem.Rules.MatchPriority;
+                    var bestMatchPriority = bestResult.CompletionItem.Rules.MatchPriority;
 
-                // Otherwise use the chosen item that has the highest matchPriority.
-                for (int i = 1, n = chosenItems.Count; i < n; i++)
-                {
-                    var chosenItem = chosenItems[i];
-                    var bestItemPriority = bestItem.Rules.MatchPriority;
-                    var currentItemPriority = chosenItem.Rules.MatchPriority;
+                    if (currentMatchPriority != bestMatchPriority)
+                    {
+                        if (currentMatchPriority > bestMatchPriority)
+                        {
+                            bestResult = currentResult;
+                        }
+
+                        continue;
+                    }
 
-                    if ((currentItemPriority > bestItemPriority) ||
-                        ((currentItemPriority == bestItemPriority) && !bestItem.IsPreferredItem() && chosenItem.IsPreferredItem()))
+                    // final preference is match to FilterText over AdditionalFilterTexts
+                    if (bestResult.MatchedWithAdditionalFilterTexts && !currentResult.MatchedWithAdditionalFilterTexts)
```

Comment:
This is another behavior change in this PR. if everything else considered equal (see comments in the code above), we will select item that matches `FilterText` over the one matches `AdditionalFilterTexts`.